uaclient/config.py

```diff
@@ -212,6 +213,18 @@ def entitlements(self):
             self._entitlements[entitlement_name] = entitlement_cfg
         return self._entitlements
 
+    @property
+    def contract_expiry_datetime(self) -> "datetime":
+        if not self._contract_expiry_datetime:
+            contractInfo = self.machine_token["machineTokenInfo"][
```

Comment:
@lucasmoura,this will break on unattached machines because machine_token is None
This will also break for non-root users because they won't have access to machine-token readability in general, should we instead be using expires" key from /var/lib/ubuntu-advantage/status.json if exists as that is world readable.

```bash
TypeError: 'NoneType' object is not subscriptable
```

Reply:
Will take this need to a separate branch for messaging that will need to check attach status prior to calling this method.